app/controllers/barbeque/sns_subscriptions_controller.rb

```diff
@@ -0,0 +1,42 @@
+class Barbeque::SnsSubscriptionsController < Barbeque::ApplicationController
+  def index
+    @sns_subscriptions = Barbeque::SNSSubscription.all
+  end
+
+  def show
+    @sns_subscription = Barbeque::SNSSubscription.find(params[:id])
+  end
+
+  def new
+    @sns_subscription = Barbeque::SNSSubscription.new
+  end
+
+  def edit
+    @sns_subscription = Barbeque::SNSSubscription.find(params[:id])
+  end
+
+  def create
+    @sns_subscription = Barbeque::SNSSubscription.new(params.require(:sns_subscription).permit(:topic_arn, :job_queue_id, :job_definition_id))
+
+    if @sns_subscription.save
```

Comment:
Why don't you create actual SNS subscription here? We [already do it for SQS](https://github.com/cookpad/barbeque/blob/7f55bf71181e556f04ce7631ddee8213eecaa253/app/controllers/barbeque/job_queues_controller.rb#L49-L57).

As far as I know, we can configure IAM properly for SQS specifying resource prefix like "barbeque-*" (but actually I didn't configure it in your company). I guess we can also do it for SNS.

But in that case, we need some authorization in Barbeque layer. So, for first step, manual operation may be good to avoid authorization necessity for now.